src/colmap/sfm/incremental_mapper.h

```diff
@@ -242,7 +242,12 @@ class IncrementalMapper {
   size_t FilterImages(const Options& options);
   size_t FilterPoints(const Options& options);
 
-  const Reconstruction& GetReconstruction() const;
+  // Getter functions
+  const std::shared_ptr<class Reconstruction>& Reconstruction() const;
+  const std::shared_ptr<IncrementalTriangulator>& Triangulator() const;
```

Comment:
Shouldn't we return shared pointers by value instead of const reference? https://stackoverflow.com/questions/10643563/how-to-return-smart-pointers-shared-ptr-by-reference-or-by-value

Reply:
Sure. This is done.